src/app/dataset-block/metadata-block/components/event-details/components/set-polling-source-event/set-polling-source-event.component.html

```diff
@@ -0,0 +1,202 @@
+<div class="d-flex justify-content-end">
+    <span class="me-2">Show as yaml</span>
+    <app-toggle
+        [isToggled]="isYamlView"
+        (toggled)="onToggleView($event)"
+    ></app-toggle>
+</div>
+
+<div *ngIf="isYamlView" class="layout lh-1 font-monospace mt-4">
+    <p class="color-key">fetch:</p>
+    <p>
+        &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;kind:
+        <span class="value-color">
+            {{ event.fetch.__typename }}
+        </span>
+    </p>
+    <ng-container *ngIf="fetchStepUrl">
+        <p>
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;url:
+            <a [href]="fetchStepUrl.url" [title]="fetchStepUrl.url"
+                ><span class="value-color lh-sm">{{
+                    fetchStepUrl.url
+                }}</span></a
+            >
+        </p>
+        <p *ngIf="fetchStepUrl.eventTime">
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;eventTime:
+        </p>
+    </ng-container>
+    <ng-container *ngIf="isFetchStepFilesGlob">
+        <p>
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;path:
+            <span class="value-color">{{ fetchStepFilesGlob.path }}</span>
+        </p>
+    </ng-container>
+    <p class="color-key">read:</p>
+    <p>
+        &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;kind:
+        <span class="value-color">{{ event.read.__typename }}</span>
+    </p>
+    <ng-container *ngIf="isReadStepCsv">
+        <p>
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;separator:
+            <span class="value-color"> {{ readStepCsv.separator }}</span>
+        </p>
+        <p>
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;header:
+            <span class="value-color"> {{ readStepCsv.header }}</span>
+        </p>
+    </ng-container>
+    <p class="color-key">merge:</p>
+    <ng-container *ngIf="isMergeStrategyLedger">
+        <p>
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;kind:
+            <span class="value-color"> {{ event.merge.__typename }}</span>
```

Comment:
Can we encapsulate all these `__typename` low level stuff in the component code behind properties?